app/services/topic_builder.rb

```diff
@@ -47,7 +47,7 @@ def parallel_subjects(subjects)
   def flat_topic(value)
     if value.parallelValue.present?
       value.parallelValue.flat_map { |topic| flat_topic(topic) }
-    elsif remove_trailing_punctuation?
+    elsif remove_trailing_punctuation? && value.value
       # comma, semicolon, and backslash are dropped
       Array(value.value.sub(/[ ,;\\]+$/, ''))
```

Comment:
For your consideration. A slightly more idiomatic way to handle this would be to use the safe navigation operator: `Array(value.value&.sub(/[ ,;\\]+$/, ''))`